content/en/docs/concepts/workloads/controllers/statefulset.md

```diff
@@ -301,6 +301,69 @@ already attempted to run with the bad configuration.
+and how PVCs are deleted during the lifecycle of a StatefulSet. You must enable the
+`StatefulSetAutoDeletePVC` [feature gate](/docs/reference/command-line-tools-reference/feature-gates/)
+to use this field. Once enabled, there are two policies you can configure for each
+StatefulSet:
+
+`whenDeleted`
+: configures the volume retention behavior that applies when the StatefulSet is deleted
+
+`whenScaled`
+: configures the volume retention behavior that applies when the replica count of
+  the StatefulSet   is reduced; for example, when scaling down the set.
+  
+For each policy that you can configure, you can set the value to either `Delete` or `Retain`.
+
+`Delete`
+: The PVCs created from the StatefulSet `volumeClaimTemplate` are deleted, for
+  each Pod affected by the policy. With the `whenDeleted` policy, as all Pods are
+  deleted when the Statefulset is deleted, all PVCs from the `volumeClaimTemplate`
+  are deleted. With the `whenScaled` policy, only PVCs corresponding to Pod replicas being scaled
+  down are deleted.
+
+`Retain` (default)
+: PVCs from the `volumeClaimTemplate` are not affected when their Pod is
+  deleted. This is the behavior before this new feature.
+
+Bear in mind that these policies **only** apply when Pods are being removed due to the
+StatefulSet being deleted or scaled down. For example, if a Pod associated with a StatefulSet
+fails due to node failure, and the control plane creates a replacement Pod, the StatefulSet
+retains the existing PVC.  The existing volume is unaffected, and the cluster will attach it to
+the node where the new Pod is about to launch.
+  
+The default for policies is `Retain`, matching the StatefulSet behavior before this new feature.
+
+Here is an example policy.
+
+```yaml
+apiVersion: apps/v1
+kind: StatefulSet
+...
+spec:
+  persistentVolumeClaimRetentionPolicy:
+    whenDeleted: Retain
+    whenScaled: Delete
+...
+```
+
+The StatefulSet {{<glossary_tooltip text="controller" term_id="controller">}} adds
+[owner references](/docs/concepts/overview/working-with-objects/owners-dependents/#owner-references-in-object-specifications) to its PVCs, which are then deleted by the
+{{<glossary_tooltip text="garbage collector" term_id="garbage-collection">}} at the appropriate time.
+When you set the `whenDeleted` policy to `Delete`, an owner reference to the StatefulSet instance
+is placed on all PVCs associated with that StatefulSet.
+
+The `whenScaled` policy is a little more complicated, because it adds an owner reference to the
+PVC’s pod, and you usually don’t want Pod deletions to remove the associated PVCs. This means that the
+StatefulSet controller must wait for scale-down events (when the StatefulSet replica count is changed
+to be less than the number of replicas currently in the cluster), and then sets pod owner references
+only on the specific Pods that are due to be terminated due to the scale-down action.
```

Comment:
Huh... reading this sounds complicated :).  Does this mean that the stateful set controller during a restart can "lose" a scale down event?  Or is the event caught because there is a PVC still there when the controller restarts?   As a user, I need to know whether force deleting a pod while the controller is down will result in the PVC being left around or not (probably as an api reviewer too :)), and this paragraph doesn't quite give clarity on that.

Reply:
Clarified, hopefully.